src/rvs_blas.cpp

```diff
@@ -480,22 +610,70 @@ bool rvs_blas::run_blass_gemm(std::string ops_type) {
       } else {
         return true;
       }
+    }
+
+    if(data_type == "fp8_r") {
+
+      rocblas_datatype a_type = rocblas_datatype_f8_r;
+      rocblas_datatype b_type = rocblas_datatype_f8_r;
+      rocblas_datatype c_type = rocblas_datatype_f8_r;
+      rocblas_datatype d_type = rocblas_datatype_f8_r;
+
+      rocblas_computetype compute_type = rocblas_compute_type_f32;
+      rocblas_gemm_algo algo = rocblas_gemm_algo_standard;
+      int32_t sol_index = 0;
+      uint32_t flags = 0;
 
-      /*
-         if (rocblas_gemm_ex(blas_handle, transa, transb,
-         rvs_blas::m, rvs_blas::n, rvs_blas::k,
-         &alpha, dhlfa, a_type, blas_lda_offset,
-         dhlfb, b_type, blas_ldb_offset, &beta,
-         dhlfc, c_type, blas_ldc_offset,
-         dhlfc, c_type, blas_ldc_offset,
-         compute_type, algo, sol_index, flags) != rocblas_status_success) {
-         is_error = true;  // GPU cannot enqueue the gemm
-         std::cout << "\n Error in Hgemm " << "\n";
-         return false;
-         } else {
-         return true;
-         }
-         */
+      _Float32 alpha = (_Float32)blas_alpha_val;
+      _Float32 beta = (_Float32)blas_beta_val;
+
+      if (rocblas_gemm_ex3(blas_handle, transa, transb,
+            rvs_blas::m, rvs_blas::n, rvs_blas::k, &alpha,
+            dda, a_type, blas_lda_offset,
+            ddb, b_type, blas_ldb_offset, &beta,
+            ddc, c_type, blas_ldc_offset,
+            ddd, d_type, blas_ldc_offset,
+            compute_type, algo, sol_index, flags) != rocblas_status_success) {
+
+        is_error = true;  // GPU cannot enqueue the gemm
+        std::cout << "\n Error in rocblas_gemm_ex3() !!! " << "\n";
+        return false;
+
+      } else {
+        return true;
+      }
+    }
+
+    if(data_type == "bf16_r") {
+
+      rocblas_datatype a_type = rocblas_datatype_bf16_r;
+      rocblas_datatype b_type = rocblas_datatype_bf16_r;
+      rocblas_datatype c_type = rocblas_datatype_bf16_r;
+      rocblas_datatype d_type = rocblas_datatype_bf16_r;
+
+      rocblas_datatype compute_type = rocblas_datatype_f32_r;
+      rocblas_gemm_algo algo = rocblas_gemm_algo_standard;
+      int32_t sol_index = 0;
+      uint32_t flags = 0;
+
+      _Float32 alpha = (_Float32)blas_alpha_val;
+      _Float32 beta = (_Float32)blas_beta_val;
+
+      if (rocblas_gemm_ex(blas_handle, transa, transb,
+            rvs_blas::m, rvs_blas::n, rvs_blas::k, &alpha,
+            dda, a_type, blas_lda_offset,
+            ddb, b_type, blas_ldb_offset, &beta,
+            ddc, c_type, blas_ldc_offset,
+            ddd, d_type, blas_ldc_offset,
+            compute_type, algo, sol_index, flags) != rocblas_status_success) {
+
+        is_error = true;  // GPU cannot enqueue the gemm
+        std::cout << "\n Error in rocblas_gemm_ex() !!!" << "\n";
```

Comment:
i see you are following prints as in file, cout is race-prone compared to rvs::log , if not in this draft, have a todo to replace?

Reply:
Yes, we have to move to rvs::log completely. There other modules as well.